float32[] pos
---
int32 err
